src/app/client/src/app/modules/groups/components/group-header/group-header.component.ts

```diff
@@ -87,6 +75,44 @@ export class GroupHeaderComponent implements OnInit, OnDestroy {
     this.groupService.addTelemetry(id, this.activatedRoute.snapshot, []);
   }
 
+  toggleModal(visibility = false, name?: string) {
+    this.showModal = visibility;
+    this.groupService.emitMenuVisibility('group');
+    this.modalName = name;
+    switch (name) {
+      case 'delete':
```

Comment:
Maintain the enum for all these status
const actions = {
  'delete': {name: "", error:"", title:""},
  'deActivate': {name: "", error:"", title:""},
  'activate': {name: "", error:"", title:""}

Try to avoid using strings at many places & conditions specific to string. 

Should try to avoid string checks.

Reply:
This is just to show modal based on which button the user has clicked.